ProjectsNetwork/Areas/User/Controllers/SkillsController.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using ProjectsNetwork.Models;
+using ProjectsNetwork.Services.IServices;
+
+// For more information on enabling MVC for empty projects, visit https://go.microsoft.com/fwlink/?LinkID=397860
+
+namespace ProjectsNetwork.Controllers
+{
+
+    [Route("User/Skills",
+        Name = "skill")]
+    [Area("User")]
+    public class SkillsController : Controller
+    {
+
+        private readonly ISkillsService _skillsService;
+
+        public SkillsController(ISkillsService skillsService)
+        {
+            this._skillsService = skillsService;
+        }
+
+        // GET: /<controller>/
+        public IActionResult Index()
+        {
+            var skills = this._skillsService.GetAll();
+            ClaimsPrincipal currentUser = this.User;
+            var currentUserID = currentUser.FindFirst(ClaimTypes.NameIdentifier).Value;
+            if (currentUserID == null)
+            {
+                return NotFound("User not found");
+            }
+            var mySkills = this._skillsService.GetMySkills(currentUserID);
+            var tupleModel = new Tuple<List<Skill>, List<Skill>, Skill>((List<Skill>)skills, mySkills, new Skill());
+            return View(tupleModel);
+        }
+
+        [HttpPost]
+        [Route("Post")]
+        public IActionResult Post(int[] skills)
```

Comment:
Again, choose better names. I don't need to be going to code to read the difference between AddSkill and Post. It needs to be something like PostUserSkill and PostSkill